backend/database_handler/migration/versions/99d3f1bc5a08_add_hash.py

```diff
@@ -0,0 +1,101 @@
+"""add_hash
+
+Revision ID: 99d3f1bc5a08
+Revises: f9636f013003
+Create Date: 2024-09-05 17:03:30.743557
+
+"""
+
+from typing import Sequence, Union
+
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.dialects import postgresql
+
+# revision identifiers, used by Alembic.
+revision: str = "99d3f1bc5a08"
+down_revision: Union[str, None] = "f9636f013003"
+branch_labels: Union[str, Sequence[str], None] = None
+depends_on: Union[str, Sequence[str], None] = None
+
+
+# This migration adds a hash column to the transactions table (instead of id) and a transaction_hash column to the transactions_audit table (instead of transaction_id).
+# To remain backwards compatible and downgradable, the hash and nonce are set to the current id on existing transactions.
+def upgrade() -> None:
+    # Add the new 'hash' column
+    op.add_column(
+        "transactions", sa.Column("hash", sa.String(length=255), nullable=True)
+    )
+
+    # Update the 'hash' column with the current 'id' value
+    op.execute("UPDATE transactions SET hash = CAST(id AS VARCHAR(255))")
```

Comment:
Doesn't this hash need to coincide with the hash logic from `TransactionsProcessor._generate_transaction_hash`?

Reply:
I replicated the hash generation logic in the migration script, however:
- Had to add some eth packages to the requirements of the db handler, is that OK?
- It effectively generates hash for legacy transactions, but it's not mirrored in the frontend. I played around with updating the legacy TX's in the frontend but it's a bit of a rabbit hole; maybe we can leave the frontend Txs without hashes and if people have issue they can just clear the local storage.

Reply:
I wouldn't care about transactions hash migration in the frontend. We could do something temporary like showing the ID for the ones that doesn't have hash

Reply:
I thought the same. What I did was migrating them from **id: 1** to **hash: 0x1** just to populate the UI. There might be edge cases with people who had pending transactions during the upgrade, but I think it would be going too far to handle thoses - worst case they can simply clear their local transactions.